File: scm-csharp/scheme/SchemeCallFrame.cs
namespace scheme;

public class SchemeCallFrame
{
    public string? name;
    public SourcePos? pos;

    public SchemeCallFrame(string? name, SourcePos? pos)
    {
        this.name = name;
        this.pos = pos;
    }

    public override string ToString()
    {
        string n = name ?? "?";
        return pos != null ? "  at " + n + " (" + pos + ")" : "  at " + n;
    }
}

File: scm-java/src/main/scheme/SchemeCallFrame.java
package scheme;

public class SchemeCallFrame {
    public String name;
    public SourcePos pos;

    public SchemeCallFrame(String name, SourcePos pos) {
        this.name = name;
        this.pos = pos;
    }

    @Override
    public String toString() {
        String n = name != null ? name : "?";
        return pos != null ? "  at " + n + " (" + pos + ")" : "  at " + n;
    }
}
